cmd/api/main.go

```diff
@@ -140,10 +148,33 @@ func main() {
 	if err != nil {
 		log.Fatalf("Failed to create server: %v", err)
 	}
+
+	// Shared options for the logger, with a custom gRPC code to log level function.
+	zapOpts := []grpc_zap.Option{
+		grpc_zap.WithDurationField(func(duration time.Duration) zapcore.Field {
+			return zap.Int64("grpc.time_duration_in_ns", duration.Nanoseconds())
```

Comment:
A bit of a nit - is nanos the right level of precision? Wouldn't milliseconds make more sense for logging (and eventually metrics?)

```suggestion
			return zap.Int64("grpc.time_duration_in_ms", duration.Milliseconds())
```